cumulusci/core/keychain/base_project_keychain.py

```diff
@@ -201,64 +206,136 @@ def get_org(self, name: str):
     def _get_org(self, name):
         return self.orgs.get(name)
 
-    def _raise_org_not_found(self, name):
-        raise OrgNotFound(f"Org named {name} was not found in keychain")
+    # This implementation of get_default_org, set_default_org, and unset_default_org
+    # is currently kept for backwards compatibility, but EncryptedFileProjectKeychain
+    # now stores the default elsewhere, and EnvironmentProjectKeychain doesn't actually
+    # persist across multiple invocations of cci, so we should consider getting rid of this.
+
+    def get_default_org(self):
+        """ retrieve the name and configuration of the default org """
+        for org in self.list_orgs():
+            org_config = self.get_org(org)
+            if org_config.default:
+                return org, org_config
+        return None, None
+
+    def remove_org(self, name, global_org=None):
+        if name in self.orgs.keys():
+            self._remove_org(name, global_org)
+        cleanup_org_cache_dirs(self, self.project_config)
+
+    def _remove_org(self, name, global_org):
+        del self.orgs[name]
+        self._load_orgs()
 
     def list_orgs(self):
         """ list the orgs configured in the keychain """
         orgs = list(self.orgs.keys())
         orgs.sort()
         return orgs
 
-    def set_service(self, name, service_config, project=False):
-        """ Store a ServiceConfig in the keychain """
-        if not self.project_config.services or name not in self.project_config.services:
-            self._raise_service_not_valid(name)
-        self._validate_service(name, service_config)
-        self._set_service(name, service_config, project)
+    def _raise_org_not_found(self, name):
+        raise OrgNotFound(f"Org named {name} was not found in keychain")
+
+    #######################################
+    #              Services               #
+    #######################################
+
+    def _load_services(self):
+        pass
+
+    def _init_default_services(self):
+        pass
+
+    def set_service(self, service_type, alias, service_config, project=False):
```

Comment:
@davisagli - Looks like this needs to be present as no other keychain class currently overrides this function. If we're going to merge EncryptedFileProjectKeychain with BaseEncryptedProjectKeychain, I could add a new `set_service()` method there that accepts the `project` kwarg.

Reply:
@Br4nd0R I thought we decided all services would be stored at the global level and only defaults would be overridable at the project level. With that in mind, what's the purpose of the project flag here?

Reply:
You are correct; all services are stored at the global level. Since this is the only place where the `set_service()` method is defined, then when it is invoked from an instance of `EncryptedFileProjectKeychain` we still need to be able to pass in this argument so that the keychain knows that we need to write out the new project level default service into the local project's `DEFAULT_SERVICES.json` file in `EncryptedFileProjectKeychain._save_default_services()`.

But I think that we can remove the project argument from both `BaseEncryptedProjectKeychain` and `EncryptedFileProjectKeychain`'s `_set_encrypted_service()` methods.

Reply:
I think if a new service is being set as the default because it's the first service of a type, we always want to set that at the global level, so don't need the project flag here. Explicitly setting a project-level override of the default can be done by a separate call to set_default_service.